core/templates/domain/skill/skill-rights-backend-api.service.spec.ts

```diff
@@ -16,127 +16,103 @@
  * @fileoverview Unit tests for SkillRightsBackendApiService.
  */
 
-// TODO(#7222): Remove the following block of unnnecessary imports once
-// the code corresponding to the spec is upgraded to Angular 8.
-import { UpgradedServices } from 'services/UpgradedServices';
-// ^^^ This block is to be removed.
-require('App.ts');
-require('domain/skill/skill-rights-backend-api.service.ts');
-require('pages/skill-editor-page/skill-editor-page.controller.ts');
-require('services/csrf-token.service.ts');
-
-describe('Skill rights backend API service', function() {
-  var SkillRightsBackendApiService = null;
-  var $rootScope = null;
-  var $scope = null;
-  var $httpBackend = null;
-  var CsrfService = null;
-
-  beforeEach(angular.mock.module('oppia'));
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    var ugs = new UpgradedServices();
-    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
-      $provide.value(key, value);
-    }
-  }));
-
-  beforeEach(angular.mock.inject(function($injector, $q) {
-    SkillRightsBackendApiService = $injector.get(
-      'SkillRightsBackendApiService');
-    $rootScope = $injector.get('$rootScope');
-    $scope = $rootScope.$new();
-    $httpBackend = $injector.get('$httpBackend');
-    CsrfService = $injector.get('CsrfTokenService');
-
-    spyOn(CsrfService, 'getTokenAsync').and.callFake(function() {
-      var deferred = $q.defer();
-      deferred.resolve('sample-csrf-token');
-      return deferred.promise;
+import { HttpClientTestingModule, HttpTestingController } from
+  '@angular/common/http/testing';
+import { TestBed, fakeAsync, flushMicrotasks, async } from
+  '@angular/core/testing';
+
+import { ISkillRightBackend } from 'domain/skill/SkillRightsObjectFactory.ts';
+import { SkillRightsBackendApiService} from
+  'domain/skill/skill-rights-backend-api.service.ts';
+
+describe('Skill rights backend API service', () => {
+  let skillRightsBackendApiService:SkillRightsBackendApiService = null;
+  let httpTestingController: HttpTestingController = null;
+  let sampleSkillRights:ISkillRightBackend = {
+    skill_id: '0',
+    can_edit_skill_description: true
+  };
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
     });
-  }));
-
-  afterEach(function() {
-    $httpBackend.verifyNoOutstandingExpectation();
-    $httpBackend.verifyNoOutstandingRequest();
+    httpTestingController = TestBed.get(HttpTestingController);
+    skillRightsBackendApiService = TestBed.get(SkillRightsBackendApiService);
+  });
+  afterEach(() => {
+    httpTestingController.verify();
   });
 
-  it('should successfully fetch a skill dict from backend', function() {
-    var successHandler = jasmine.createSpy('success');
-    var failHandler = jasmine.createSpy('fail');
-    var sampleResults = {
-      skill_id: '0',
-      can_edit_skill_description: ''
-    };
+  it('should report a cached skill rights after caching it',
```

Comment:
Please do not change the order of tests. It becomes difficult to review the changes.

Reply:
Earlier only one test was there then more test was added in that so i added them after the existing test .
I'll fix them 
Thanks

Reply:
changed the order of tests